### PR TITLE
cli: parse http-proxy and https-proxy

### PR DESCRIPTION
**Pull request by mr-cal**
- [X] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [X] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [X] Have you successfully run `make lint`?
- [X] Have you successfully run `pytest tests/unit`?

-----
Parse the command line arguments `--http-proxy` and `--https-proxy`.

source: https://forum.snapcraft.io/t/error-unrecognized-arguments-http-proxy/30967

(CRAFT-1217)

### CODE REVIEW

File: tests/unit/cli/test_lifecycle.py [hunk 1]
```diff
@@ -640,11 +748,17 @@ def test_lifecycle_command_pack_build_for(mocker):
...
                 target_arch=None,
                 provider=None,
+                http_proxy=None,
+                https_proxy=None,
             )
         )
     ]
 
 
+def test_lifecycle_command_http_proxy(mocker):
```
Comment (sergiusens) [inline tests/unit/cli/test_lifecycle.py:758]: what's this for?

**Comment by sergiusens** [comment]
This seems to enable psasthrough, but I thought we were going to enable the proxy for core22 as well

File: snapcraft/commands/lifecycle.py [hunk 2]
```diff
@@ -74,6 +74,18 @@ def fill_parser(self, parser: "argparse.ArgumentParser") -> None:
...
+            type=str,
+            default=os.getenv("http_proxy"),
+            help="Set http proxy",
+        )
+        parser.add_argument(
+            "--https-proxy",
+            type=str,
+            default=os.getenv("https_proxy"),
+            help="Set https proxy",
+        )
```
Comment (cmatsuoka) [inline snapcraft/commands/lifecycle.py:88]: Where are these values used? Are we just allowing the user to define them? (If the actual implementation is just setting the environment variable value and allowing it to pass to the inner instance I think we should do that)

Comment (sergiusens) [reply snapcraft/commands/lifecycle.py:88]: this is for the legacy execution to work. I believe I will let this in, but we need a task to enable it for core22

Comment (mr-cal) [reply snapcraft/commands/lifecycle.py:88]: Is this what you mean by "enabling the proxy"?
https://github.com/snapcore/snapcraft/blob/260214788af30ea055b3170afd97b91e556341ae/snapcraft_legacy/internal/build_providers/_base_provider.py#L510

File: tests/unit/cli/test_lifecycle.py (hunk 1 above)
Comment (mr-cal) [reply tests/unit/cli/test_lifecycle.py]: whoops, removed

**Review by sergiusens — APPROVED** [review]

**Pull request merged**